farm/data_handler/processor.py

```diff
@@ -346,14 +347,23 @@ def _init_samples_in_baskets(self):
                 logger.error(f"Error message: {e}")
 
     def _featurize_samples(self):
+        curr_problematic_sample_ids = []
         for basket in self.baskets:
             for sample in basket.samples:
                 try:
                     sample.features = self._sample_to_features(sample=sample)
                 except Exception as e:
-                    logger.error(f"Could not convert this sample to features: \n {sample}")
-                    logger.error(f"Basket id: id_internal: {basket.id_internal}, id_external: {basket.id_external}")
-                    logger.error(f"Error message: {e}")
+                    curr_problematic_sample_ids.append(sample.id)
+        if curr_problematic_sample_ids:
+            self.problematic_sample_ids.update(curr_problematic_sample_ids)
```

Comment:
Why are we updating the problematic_sample_ids here and when we collect results from MP?

Reply:
The idea here is that we have two levels at which `problematic_ids` are stored. Within multiprocessing, each Processor needs to collect `problematic_ids` as it iterates through the baskets in its chunks. These are eventually returned out of multiprocessing via `dataset_from_dicts( )`. At this point, these returned `problematic_ids` are [collected together in the non-multiprocessing Processor ](https://github.com/deepset-ai/FARM/blob/ca358df76bd7e6ec952d22eb9b90e3c7ae733c7a/farm/data_handler/data_silo.py#L188). 

I went for this approach because it allows for us to implement logging problematic samples neatly as a method of the processor (`Processor.log_problematic( )`) as long as the problematic sample ids are stored in the `Processor`. 

I do admit that it might be a little confusing coordinating between the higher level single `Processor` and the multiple lower level `Processor`s. Do you have any thoughts or suggestions on this?